### ISSUE TITLE
Host Kusto notebooks which query database for live data like PowerBI reports

### ISSUE THREAD

**Issue by joyceerhl** [DonJayamanne/vscode-kusto#27]
Today all our (internal) dashboards are PowerBI dashboards. PowerBI has no concept of version control, so anytime you want to update a PowerBI report, you have to:
1. Download the report 
2. Make changes in local PowerBI
3. Tell everyone on your team with push rights to PowerBI to not publish updates
4. Publish your updated report 
5. They then have to manually merge your changes with theirs (if any)

I've had to do this several times (creating/updating shared PowerBI dashboards) and it's a massive pain. 

We could host live Kusto notebooks on a webpage (this would effectively let us build version-controllable and hence collaboration-friendly PowerBI reports).

Original thought was to be able to export notebooks as PowerBI reports, but I imagine *.pbix is proprietary.

/cc @claudiaregio

**Comment by DonJayamanne** [comment]
*.pbix is just a zip file hence we should be able to create this fairly easily from within the extension. Just a thought.
There seem to be plenty of powerbi npm packages. Though, i feel that such functionality might be best served as a seprate extension.
but lets trash out the requirements first, as this sounds like a good idea.

**Comment by claudiaregio** [comment]
Yes, I had another internal employee complain about this same thing when using PowerBI and they've moved to using notebooks for all their analyses. They said the one thing missing was the ability to host these notebooks on a dashboard

**Comment by DonJayamanne** [comment]
> They said the one thing missing was the ability to host these notebooks on a dashboard

* They can use Jupyter notebooks with kqlmagic to run kusto queries
* Then host the notebooks on a dashboard using [voila](https://github.com/voila-dashboards/voila) (its kind of the industry standard, IMHO)

**Comment by claudiaregio** [comment]
@DonJayamanne do you happen to know if there is anything prohibiting us from using that today?

**Comment by DonJayamanne** [comment]
* We can host voila on azure app service

The only challenge I can see is authentication. Sure we can have app service and rest of the code running under the user credentials, sounds very risky with ms credentials. I'd never do this, specially with all of the 3rd party code running...

Should check if there's a way to do with kusto (some other auth or credentials). 
Let me check

**Comment by DonJayamanne** [comment]
Yes this is possible, nothing stopping us today.
Can use certificates or the like to authenticate on hosted server (azure app service etc)
Check auth types here https://mybinder.org/v2/gh/Microsoft/jupyter-Kqlmagic/master?filepath=notebooks%2FQuickStart.ipynb
